releases-implementation/README.md

```diff
@@ -0,0 +1,415 @@
+  - name: cloud-provider-aws
+    version: 1.25.14-gs2
+    dependsOn:
+    - vertical-pod-autoscaler-crd
+  - name: cluster-autoscaler
+    version: 1.27.3-gs8
+    dependsOn:
+    - kyverno
+  - name: coredns
+    version: 1.21.0
+  - name: external-dns
+    version: 3.1.0
+    dependsOn:
+    - prometheus-operator-crd
+  - name: metrics-server
+    version: 2.4.2
+    dependsOn:
+    - kyverno
+  - name: net-exporter
+    version: 1.19.0
+    dependsOn:
+    - prometheus-operator-crd
+  - name: network-policies
+    version: 0.1.0
+    catalog: cluster
+  - name: node-exporter
+    version: 1.19.0
+    dependsOn:
+    - kyverno
+  - name: vertical-pod-autoscaler
+    version: 5.1.0
+    dependsOn:
+    - prometheus-operator-crd
+  - name: vertical-pod-autoscaler-crd
+    version: 3.0.0
+  - name: etcd-k8s-res-count-exporter
+    version: 1.10.0
+    dependsOn:
+    - kyverno
+  - name: observability-bundle
+    version: 1.3.4
+    dependsOn:
+    - coredns
+  - name: k8s-dns-node-cache
+    version: 2.6.2
+    dependsOn:
+    - kyverno
+  - name: security-bundle
+    version: 1.6.5
+    catalog: giantswarm
+    dependsOn:
+    - prometheus-operator-crd
+  - name: teleport-kube-agent
+    version: 0.9.0
+  components:
+  - name: cluster-aws
+    version: 1.0.0
+  - name: flatcar
+    version: 3815.2.2
+  - name: flatcar-variant
+    version: 3.0.0
+  - name: kubernetes
+    version: 1.25.16
+  date: "2024-05-18T12:57:50Z"
+  state: active
+```
+
+New releases MUST BE added via pull requests in a similar way like for vintage releases. After a pull request is merged, CircleCI job pushes the newly added Releases to the releases catalog and to the provider-specific app collection.
+
+#### 4.1.2. Deploying a cluster
+
+Today workload clusters in our Cluster API-based KaaS product are deployed with cluster-$provider apps. That remains the same, with only one minor difference, which is how we specify the version, where instead of specifying App’s `.spec.version` property we MUST specify `release.giantswarm.io/version` label and leave `.spec.version` empty.
+
+So the whole manifest for the cluster-$provider app and its config would look like this:
+
+```yaml
+---
+apiVersion: v1
+data:
+  values: |
+    global:
+      metadata:
+        name: mycluster
+        organization: mycompany
+        description: Production cluster
+kind: ConfigMap
+metadata:
+  creationTimestamp: null
+  labels:
+    giantswarm.io/cluster: mycluster
+  name: mycluster-userconfig
+  namespace: org-mycompany
+---
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+metadata:
+  labels:
+    app-operator.giantswarm.io/version: 0.0.0
+    release.giantswarm.io/version: 25.0.0
+  name: mycluster
+  namespace: org-mycompany
+spec:
+  catalog: cluster
+  kubeConfig:
+    inCluster: true
+  name: cluster-aws
+  namespace: org-mycompany
+  userConfig:
+    configMap:
+      name: mycluster-userconfig
+      namespace: org-mycompany
+  version: ""
```

Comment:
Will this play nicely with gitops? Does both Flux and Argo allow this to be empty in git and not try and re-apply it to empty after the mutation has happened?

Reply:
AFAIK an empty field should not be a problem for Flux. We already leave some fields empty for some resources in our gitops setup.

Will check this though, thanks!